hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -1280,4 +1281,10 @@ private Set<String> getDropPartitionColNames() {
     }
     return new HashSet<>(Arrays.asList(partitionFields.get()));
   }
+
+  public ReaderContextFactory<T> getReaderContextFactoryForWrite() {
+    // question: should we just return null when context is serialized as null? the mismatch reader context would throw anyway.
+    return (ReaderContextFactory<T>) getContext().getReaderContextFactoryForWrite(metaClient, config.getRecordMerger().getRecordType(),
```

Comment:
I feel the spark reader context usage in write path as a one off thing. and we can leave it as is. apart from this use-case, we don't forsee more use-cases in near future.